src/Uno.Extensions.Reactive.UI/Presentation/Bindings/Collections/Data/DataLayer.cs

```diff
@@ -0,0 +1,228 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Uno.Extensions.Collections;
+using Uno.Extensions.Reactive.Bindings.Collections._BindableCollection.Facets;
+
+#if WINUI
+using ISchedulerInfo = Microsoft.UI.Dispatching.DispatcherQueue;
+#else
+using ISchedulerInfo = Windows.System.DispatcherQueue;
+#endif
+
+namespace Uno.Extensions.Reactive.Bindings.Collections._BindableCollection.Data
+{
+	/// <summary>
+	/// A holder of a data layer for a given scheduler context
+	/// </summary>
+	internal sealed class DataLayer : ILayerHolder, IBindableCollectionViewSource, IDisposable
+	{
+		private readonly DataLayer? _parent;
+		private readonly ISchedulerInfo? _context;
+		private readonly IBindableCollectionDataLayerStrategy _layerStrategy;
+		private readonly IEnumerable<object> _facets;
+
+		private DataLayerChangesBuffer? _currentChangesBuffer;
+		private DataLayerChangesBuffer? _nextChangesBuffer;
+
+		/// <summary>
+		/// The view of the <see cref="Items"/> that can be used for data bindings.
+		/// </summary>
+		public ICollectionView View { get; }
+
+		/// <summary>
+		/// The source Items of the <see cref="View"/>.
+		/// </summary>
+		public CollectionFacet Items { get; }
+
+		//public IObservableCollection? CurrentSource => _currentChangesBuffer?.Collection;
+
+		public IBindableCollectionViewSource? Parent => _parent;
+
+		//public IScheduler Scheduler => _parent != null
+		//	? _parent.Scheduler
+		//	: (_context.IsOnDispatcher
+		//		? _context.GetDispatcher()
+		//		: System.Reactive.Concurrency.Scheduler.Immediate);
+
+		/// <summary>
+		/// Creates a holder for the root layer of data
+		/// </summary>
+		public static DataLayer Create(
+			IBindableCollectionDataLayerStrategy layerStrategy,
+			IObservableCollection items, 
+			ISchedulerInfo? context)
+		{
+			var holder = new DataLayer(null, layerStrategy, context);
+			var initContext = layerStrategy.CreateUpdateContext(VisitorType.InitializeCollection, TrackingMode.Reset);
+			var initializer = holder.Init(items, initContext);
+
+			// We need to Flush buffer even for a Init: even if we don't have a CollectChange to raise, we may have other (applicative) callbacks.
+			holder.Schedule(initializer.Complete);
+
+			return holder;
+		}
+
+		/// <summary>
+		/// Creates a holder for a sub data layer of this
+		/// </summary>
+		public (DataLayer holder, DataLayerUpdate initializer) CreateSubLayer(IObservableCollection subItems, IUpdateContext changes)
```

Comment:
Would it be best to use records instead of tuples for public apis (even if the type is internal)?

Reply:
I would agree. I'll probably change it in a future commit, but as it's an internal class I'll keep it like this to reduce amount of changes.